pkg/reconciler/ingress/lister.go

```diff
@@ -73,23 +73,35 @@ func (l *gatewayPodTargetLister) ListProbeTargets(ctx context.Context, ing *v1al
 		if err != nil {
 			return nil, fmt.Errorf("failed to get Gateway %q: %w", gatewayName, err)
 		}
-		targets, err := l.listGatewayTargets(gateway)
+		targets, excludedHosts, err := l.listGatewayTargets(gateway)
 		if err != nil {
 			return nil, fmt.Errorf("failed to list the probing URLs of Gateway %q: %w", gatewayName, err)
 		}
+
 		if len(targets) == 0 {
 			continue
 		}
 		for _, target := range targets {
+			// Pick a single host since they all end up being used in the same
+			// VirtualService and will be applied atomically by Istio. However,
+			// it's important to choose a hostname which isn't excluded from the
+			// initial list of gateway targets.
+			var host string
+			for _, hostName := range hostsByGateway[gatewayName].List() {
+				if !excludedHosts.Has(hostName) {
```

Comment:
If `myapp.example.com` has `tls.HttpsRedirect`, prober will exclude the host from target even with `HTTPS`?

Reply:
Good point. I've removed the redirected hosts from the list of excluded hosts.